### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by anatolii-zarytskyi**

### CODE REVIEW

File: src/main/java/core/stack/MateStack.java [hunk 1]
```diff
@@ -1,19 +1,44 @@
...
+        if (array.length == size) {
+            grow();
+        }
+        array[size++] = value;
     }
 
     public T peek() {
-        return null;
+        int len = size();
+        if (len == 0) {
```
Comment (mankokolya) [inline src/main/java/core/stack/MateStack.java:23]: Зачем тебе len и size()  использую size

**Review by mankokolya — CHANGES_REQUESTED** [review]

File: src/main/java/core/stack/MateStack.java [hunk 2]
```diff
@@ -1,19 +1,44 @@
...
+        size--;
+        return object;
     }
 
     public int size() {
-        return 0;
+        return size;
+    }
+
+    private T[] grow() {
```
[the remark this answers is not in the document]
Comment (Anatolii-Zarytskyi) [reply src/main/java/core/stack/MateStack.java]: виправив

File: src/main/java/core/stack/MateStack.java (hunk 1 above)
Comment (Anatolii-Zarytskyi) [reply src/main/java/core/stack/MateStack.java]: виправив

File: src/main/java/core/stack/MateStack.java [hunk 3]
```diff
@@ -1,19 +1,44 @@
...
+        T[] grewArray = (T[]) new Object[size + (size / 2)];
+        System.arraycopy(array, 0, grewArray, 0, size);
```
[the remark this answers is not in the document]
Comment (Anatolii-Zarytskyi) [reply src/main/java/core/stack/MateStack.java]: цікавий підхід

**Review by mankokolya — APPROVED** [review]

File: src/main/java/core/stack/MateStack.java [hunk 4]
```diff
@@ -1,19 +1,42 @@
...
-        return null;
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
+        return (T) array[size - 1];
     }
 
     public T pop() {
-        return null;
+        T object = peek();
```
Comment (Max-Pochepets) [inline src/main/java/core/stack/MateStack.java:34]: did the value became `null` after peek? or we still have it in stack?

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

File: src/main/java/core/stack/MateStack.java (hunk 4 above)
Comment (Anatolii-Zarytskyi) [reply src/main/java/core/stack/MateStack.java:34]: fixed it

**Review by Max-Pochepets — APPROVED** [review]

**Pull request closed without merging**